shelley/chain-and-ledger/shelley-spec-ledger-test/test/Test/Shelley/Spec/Ledger/UnitTests.hs

```diff
@@ -705,6 +707,25 @@ testPoolCostTooSmall =
             )
         }
 
+testProducedOverMaxWord64 :: Assertion
+testProducedOverMaxWord64 =
+  let biggestCoin = fromIntegral (maxBound :: Word64)
+      txbody =
+        TxBody @C
+          (Set.fromList [TxIn genesisId 0])
+          (StrictSeq.fromList [TxOut bobAddr (Coin biggestCoin)])
+          Empty
+          (Wdrl Map.empty)
+          (Coin 1) -- @produced@ will return biggestCoin + 1, which is > 2^64.
+          (SlotNo 100)
+          SNothing
+          SNothing
+      wits = mempty {addrWits = makeWitnessesVKey (hashAnnotated txbody) [alicePay]}
+      tx = Tx @C txbody wits SNothing
+      st = runShelleyBase $ applySTSTest @(LEDGER C) (TRC (ledgerEnv, (utxoState, dpState), tx))
+   -- We test that the serialization of the predicate failure does not return bottom
+   in serialize' st @?= serialize'  st
```

Comment:
is there a less silly way to enforce that the serialization does not return bottom?

Reply:
You could use `rnf`, `deepseq`, or `force` on `serialize' st`. What you have works too.

(there are two spaces before the second `st`)